### PR TITLE
[14.0][ADD] base_revision

### PR DESCRIPTION
**Pull request by kittiu**
**Note:** this module was split from sale_order_revision.

From version 14 on, to use sale_order_revision, we will need base_revision + sale_order_revision.
https://github.com/OCA/sale-workflow/pull/1369

### CODE REVIEW

**Comment by newtratip** [comment]
@kittiu, Could you check your tavis CI ?

File: base_revision/models/base_revision.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            "view_mode": "tree,form",
+            "name": _("New Revisions"),
+            "res_model": self._name,
+            "domain": "[('id', 'in', %s)]" % revision_ids,
+            "auto_search": True,
+            "target": "current",
+            "nodestroy": True,
+        }
+
+        # Returning the new sale order view with new record.
```
Comment (Saran440) [inline base_revision/models/base_revision.py]:
```suggestion
        # Returning the new revision view with new record.
```

**Review by Saran440 — APPROVED** [review]
Functional Test :+1: 

Code reviewed and please, pre-commit this PR.

File: base_revision/models/base_revision.py (hunk 1 above)
Comment (kittiu) [reply base_revision/models/base_revision.py]: Fixed.

**Comment by kittiu** [comment]
@Saran440 @newtratip ready for review.

File: base_revision/tests/test_base_revision.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+from odoo.tests import common
+
+from .base_revision_tester import BaseRevisionTester
+from .common import setup_test_model, teardown_test_model
+
+
+class TestBaseRevision(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestBaseRevision, cls).setUpClass()
```
Comment (etobella) [inline base_revision/tests/test_base_revision.py:14]: Use the fakeModelLoader: https://github.com/OCA/edi/blob/13.0/edi/tests/test_consumer_mixin.py#L24

Comment (pedrobaeza) [reply base_revision/tests/test_base_revision.py:14]: TIL

Comment (kittiu) [reply base_revision/tests/test_base_revision.py:14]: Fixed. Thanks for your note about this.

**Comment by kittiu** [comment]
Fix code coverage. ready for review.

File: base_revision/models/base_revision.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        rec = super().copy(default=default)
+        if not rec.unrevisioned_name:
+            name_field = self._context.get("revision_name_field", "name")
+            rec.write({"unrevisioned_name": rec[name_field]})
+        return rec
+
+    def copy_revision_with_context(self):
+        default_data = self.default_get([])
+        new_rev_number = self.revision_number + 1
+        default_data.update(
```
Comment (Saran440) [inline base_revision/models/base_revision.py]: Could you add hook function for this?
It should be flexible for change value.

**Review by Chandresh-OSI — APPROVED** [review]

**Pull request merged**